lms/static/scripts/frontend_apps/components/GroupConfigSelector.js

```diff
@@ -25,6 +25,88 @@ import ErrorModal from './ErrorModal';
  *   be the `id` of a `GroupSet` returned by the LMS backend.
  */
 
+/**
+ *  Labeled <select> for selecting a group set for an assignment
+ *
+ * @typedef GroupSelectProps
+ * @prop {boolean} busy - A request for groups is currently in flight
+ * @prop {GroupSet[]|null} groupSets
+ * @prop {(id: string|null) => void} onInput
+ * @prop {string|null} selectedGroupSetID
+ *
+ * @param {GroupSelectProps} props
+ */
+function GroupSelect({ busy, groupSets, onInput, selectedGroupSetID }) {
+  const selectID = useUniqueId('GroupSetSelector__select');
+
+  return (
+    <>
+      <label htmlFor={selectID}>Group set: </label>
+      <select
+        disabled={busy}
+        id={selectID}
+        onInput={e =>
+          onInput(/** @type {HTMLInputElement} */ (e.target).value || null)
+        }
+      >
+        {busy && <option>Fetching group sets…</option>}
+        {groupSets && (
+          <>
+            <option disabled selected={selectedGroupSetID === null}>
+              Select group set
+            </option>
+            <hr />
+            {groupSets.map(gs => (
+              <option
+                key={gs.id}
+                value={gs.id}
+                selected={gs.id === selectedGroupSetID}
+                data-testid="groupset-option"
+              >
+                {gs.name}
+              </option>
+            ))}
+          </>
+        )}
+      </select>
+    </>
+  );
+}
+
+/**
+ * ErrorModal shown when the fetched list of course group sets is empty
+ *
+ * @param {object} props
+ * @param {() => void} props.onCancel
+ */
+function NoGroupsError({ onCancel }) {
+  return (
+    <ErrorModal
+      onCancel={onCancel}
+      title="No group sets found"
+      withErrorDetails
```

Comment:
In this case, I want to show the help text and links that are part of `ErrorDisplay`, but I don't want `ErrorModal` to have to reason about the particulars of the `Error` object at hand. Ergo, I added a prop to `ErrorModal` to handle this.

Another way I could do this is:

```
<ErrorModal onCancel={onCancel} title="No group sets found" error={{}}> ...
```

as `ErrorModal` will automatically render `ErrorDisplay` if an `ErrorLike` object is provided for `error` (no properties on an `ErrorLike` object are required). But this seemed stinky because it didn't communicate the intent, which is: show error "details" and links but don't actually operate on a real Error object. Yes, we have a real Error we _could_ use here—the `fetchError` set on `GroupConfigSelector` is a `GrouListEmptyError`, but I want to retain precise control over the text rendered in the modal (the Error's message rendered here is distracting), so I'd rather not involve the Error itself.

Reply:
> In this case, I want to show the help text and links that are part of ErrorDisplay, but I don't want ErrorModal to have to reason about the particulars of the Error object at hand.

Testing this out manually, it seems that `withErrorDetails` is used only to show this text at the bottom of the dialog:

<img width="562" alt="support-links" src="https://user-images.githubusercontent.com/2458/152162756-203d9c66-6bc9-4c98-9c88-345bf9314cb7.png">

Is that correct? The details of the error are used to prefill the form that "open a support link" shows. Currently on this PR the prefilled error summary is generic:

<img width="810" alt="Prefilled error summary" src="https://user-images.githubusercontent.com/2458/152163841-01434563-263b-411b-8ddf-8fa387e5493c.png">

Finally, given that this dialog already has a link into the knowledge base, I wonder if we want to include these generic links as well.

Reply:
OK! I'm persuaded we don't want to display these links. That also means I can completely eliminate the commit that messes with `ErrorModal` and `ErrorDisplay`, which simplifies this changeset overall 👍🏻